tests/unit/lib/foxbox/foxbox_test.js

```diff
@@ -0,0 +1,84 @@
+import Foxbox from 'js/lib/foxbox/foxbox';
+
+describe('Foxbox >', function() {
+  let foxbox;
+  const singleBox = [{
+    public_ip: '1.1.1.1',
+    client: 'abc',
+    message: JSON.stringify({
+      local_origin: 'https://local.abc.box.knilxof.org:3000',
+      tunnel_origin: 'null',
+    }),
+    timestamp: Math.floor(Date.now() / 1000),
+  }];
+  const multiBox = singleBox.concat([
+    {
+      public_ip: '2.2.2.2',
+      client: 'def',
+      message: JSON.stringify({
+        local_origin: 'http://local.def.box.knilxof.org:3000',
+        tunnel_origin: 'null',
+      }),
+      timestamp: Math.floor(Date.now() / 1000),
+    },
+  ]);
+
+  function getMockNet(single = true) {
+    return {
+      init: sinon.stub(),
+      fetchJSON: () => Promise.resolve(single ? singleBox : multiBox),
+    };
+  }
+
+  describe('constructor >', function() {
+    it('exposes a non extensible object', function() {
+      foxbox = new Foxbox();
+
+      assert.isObject(foxbox);
+      assert.throws(() => foxbox.newProp = true);
+    });
+
+    it('a net parameter can be provided', function(done) {
+      const mockNet = getMockNet();
+      foxbox = new Foxbox({ net: mockNet });
+
+      foxbox.init()
+        .then(() => {
+          sinon.assert.calledOnce(mockNet.init);
+        })
+        .then(done, done);
+    });
+  });
+
+  describe('_initDiscovery >', function() {
```

Comment:
Ideally (always?) we don't want to test private methods as they are not part of the public API and can be refactored in any way and tests should not change if input/output is the same. It feels like what we really want to test here is public ```foxbox.init``` and specifically that:
* it calls ```fetchJSON``` with correct arguments;
* if it fails and we don't have local/remote origin from the previous time, it's called once again after interval (can be mocked with sinon.clock);
* if it succeed - box list is updated (unfortunately I've added this event in my PR that is waiting your one to land :), with this event it would be easier, but you can rely on box-online event for now);
* user session is initialized (it's even easier, but maybe not in this PR).

I know it's quite painful to test promise based code (especially when methods don't return promises that we'd like to test). We have two helper methods that should help with this - https://github.com/azasypkin/app/blob/issue-111-watchers/tests/unit/test-utils.js

What do you think?